tests/integration/test_vulnerability_detector/test_providers/test_providers_os.py

```diff
@@ -67,17 +120,36 @@ def get_configuration(request):
 
 @pytest.mark.skip(reason="It will be blocked by wazuh#9309, when it was solve we can enable again this test")
 def test_providers(get_configuration, configure_environment, restart_modulesd):
-    """Check if modulesd downloads the feeds for each os.
-
-    For this, it verifies that the update of the feeds is started by checking
-    the log file. Additionally, it verifies that a warning message is generated
-    regarding the improper use of the <os> tag in providers that do not use it.
-
-    Args:
-        get_configuration (fixture): Get configurations from the module.
-        configure_environment (fixture): Configure a custom environment for testing.
-        restart_modulesd (fixture): Reset the logs file and start a new monitor.
-    """
+    '''
+    description: Check if modulesd downloads the feeds for each os. To do this, it verifies that the update of the feeds
+                 is started by checking the log file. Additionally, it verifies that a warning message is generated
+                 regarding the improper use of the <os> tag in providers that do not use it.
+
+    wazuh_min_version: 4.2.0
+
+    parameters:
+        - get_configuration:
+            type: fixture
+            brief: Get configurations from the module.
+        - configure_environment:
+            type: fixture
+            brief: Configure a custom environment for testing.
+        - restart_modulesd:
+            type: fixture
+            brief: Reset the logs file and start a new monitor.
+
+    assertions:
+        - Verify that the provider os data update starts when `os` has a determined value.
+        - Verify that a warning message appears when using `os` tag in Red Hat and NVD providers.
+
+    input_description:
+        - Test cases are found in the test module and include parameters for the `os` tag of the provider option. The
+          `wazuh_providers_os.yaml` file provides the configuration of this module for this test.
+
+    expected_output:
+        - r'Warning log 'Invalid option 'os' for .* provider not found'
+        - r'Could not find .* update starting log'
```

Comment:
Add the following callbacks

```suggestion
        - r'.*Invalid option .* for .* provider.*'
        - r'Could not find .* update starting log'
```